### PR TITLE
Import translations in CSV

### PR DESCRIPTION
**Pull request by mart-e**
Fixes odoo/odoo#50975

### CODE REVIEW

File: odoo/tools/translate.py [hunk 1]
```diff
@@ -553,10 +553,20 @@ def __init__(self, source):
         self.source = pycompat.csv_reader(source, quotechar='"', delimiter=',')
         # read the first line of the file (it contains columns titles)
         self.fields = next(self.source)
+        self.fields += list(set(["imd_name", "imd_model"]) - set(self.fields))
```
Comment (xmo-odoo) [inline odoo/tools/translate.py:556]: Why the concatenation? Why not just
```suggestion
        self.fields = list(set(next(self.source)) | {"imd_name", "imd_model"})
```

File: odoo/tools/translate.py [hunk 2]
```diff
@@ -553,10 +553,20 @@ def __init__(self, source):
...
 
     def __iter__(self):
         for entry in self.source:
-            yield zip(self.fields, entry)
+            res = dict(zip(self.fields, entry))
```
Comment (xmo-odoo) [inline odoo/tools/translate.py:560]: Maybe convert `self.source` to a `csv.DictReader` and get rid of `self.fields` then?

File: odoo/tools/translate.py (hunk 1 above)
Comment (mart-e) [reply odoo/tools/translate.py:556]: Because order is important. If I convert `next(self.source)`, I potentially lose the order for the `zip` following

Comment (xmo-odoo) [reply odoo/tools/translate.py:556]: Ah yes, good point.

**Comment by mart-e** [comment]
robodoo r+

**Pull request closed without merging**